Feature/o auth2

sfcbetiuc opened the pull request.

Commit:
Started working on the OAuth2.0 support

Commit:
SDK client env data gather, refactor

Created the logic for adding in the user agent info about user env for of all the requests made from the SDK, created a factory for getting SUTs for tests

Commit:
Fixed misplaced User-Agent, other minor changes

Commit:
Removed typo

Commit:
Removed unused import, refactored the SUT factory

Commit:
Added unit tests for RuntimeInformationProvider

Commit:
Finished implementing OAuth2.0, started working on integration tests

Commit:
Added integration tests, minor refactoring

Added integration tests for AssetApi, CampaignApi, refactored the AuthService

Commit:
Small changes after updating the mustache files

sfadincescu commented on src/Auth/OAuth.js:
```diff
@@ -0,0 +1,18 @@
+class OAuth {
```
This object contains data from 2 different entities:

1. Data about the access token (this._accessToken)
2. Data referring to the authentication (this._type)

I would split this in 2 objects:

1. Authentication (where you have type and configuration) - The TokenResponse with little tweaks can be turned into this
2. AccessToken: this would hold the relevant information about the access token see [HERE](https://github.com/thephpleague/oauth2-client/blob/master/src/Token/AccessToken.php)

sfadincescu commented on src/Auth/TokenResponse.js:
```diff
@@ -0,0 +1,50 @@
+class TokenResponse{
```
This object is not needed. You are converting a native structure (array) into an object, thus creating memory overhead. I would change this in the Authentication object as described in the comment from the OAuth object

Commit:
Added integration tests for AuthService

Commit:
Refactored the OAuth2.0 implementation, other small changes

- Refactored the applyAuthToRequest method so that it now accepts just
the oauth2 authentication type
- Moved the initialization of the ApiClient.authentication from the
BaseApi to the ApiClient class
- Refactored the name of a class: OAuth --> Autentication because it was
misleading to have a class with the name OAuth that has property called
'type' that can have the value 'oauth2'
- Removed an unnecessary 'await' from inside of the 'getTokenResponse'
method
- Fixed typos in the TokenResponse class

Commit:
Commit adding changes missed in the previous commit

Commit:
OAuth2.0 refactor that tackles code review concerns

- Removed the unnecessary "TokenResponse" class
- Other small changes based on the class removal

Commit:
Fixed the payload of the getTokenResponse method

- Added the account_id parameter to the auth request payload

Commit:
Removed timeouts from integration tests

Commit:
Removed unused dependencies from package.json

- Removed babel and babel-cli from the dependencies section of the
package.json file as these packages were automatically added by swagger
as project dependencies, but they are not used
- Generated the new package-lock.json file

Commit:
Updated the RuntimeInformationProvider class

- Modified the format of the User-Agent sent in the request header

Commit:
Changed SDK version fetch source

- The SDK version is fetched now from package.json, not from
package-lock.json
